modules/data.remote/R/remote_process.R

```diff
@@ -0,0 +1,821 @@
+        write_raw_start        <- req_start
+        write_raw_end          <- req_end
+        raw_merge              <- FALSE
+        existing_raw_file_path <- NULL
+      }
+      if (!is.null(out_process_data)) {
+        stage_process_data     <- TRUE
+        write_pro_start        <- req_start
+        write_pro_end          <- req_end
+        pro_merge              <- FALSE
+        process_file_name      <- NULL
+        existing_pro_file_path <- NULL
+        remotefile_check_flag  <- 1
+      }
+    }
+    
+  } else{
+    # db is completely empty for the given siteid
+    PEcAn.logger::logger.info("DB is completely empty for this site")
+    remotefile_check_flag                   <- 1
+    start                  <- req_start
+    end                    <- req_end
+    stage_get_data         <- TRUE
+    write_raw_start        <- req_start
+    write_raw_end          <- req_end
+    raw_merge              <- FALSE
+    existing_raw_file_path <- NULL
+    if (!is.null(out_process_data)) {
+      stage_process_data     <- TRUE
+      write_pro_start        <- req_start
+      write_pro_end          <- req_end
+      pro_merge              <- FALSE
+      existing_pro_file_path <- NULL
+    }
+  }
+  
+  outdir <- file.path(outdir, paste(source, "site", siteid_short, sep = "_"))
+  
+  fcn.args <- list()
+  fcn.args$coords                 <- coords
+  fcn.args$outdir                 <- outdir
+  fcn.args$start                  <- start
+  fcn.args$end                    <- end
+  fcn.args$source                 <- source
+  fcn.args$collection             <- collection
+  fcn.args$siteid                 <- siteid_short
+  fcn.args$scale                  <- as.double(scale)
+  fcn.args$projection             <- projection
+  fcn.args$qc                     <- as.double(qc)
+  fcn.args$algorithm              <- algorithm
+  fcn.args$input_file             <- input_file
+  fcn.args$credfile               <- credfile
+  fcn.args$out_get_data           <- out_get_data
+  fcn.args$out_process_data       <- out_process_data
+  fcn.args$stage_get_data         <- stage_get_data
+  fcn.args$stage_process_data     <- stage_process_data
+  fcn.args$raw_merge              <- raw_merge
+  fcn.args$pro_merge              <- pro_merge
+  fcn.args$existing_raw_file_path <- existing_raw_file_path
+  fcn.args$existing_pro_file_path <- existing_pro_file_path
+  
+  
+  arg.string <- PEcAn.utils::listToArgString(fcn.args)
+  
+  cmdFcn <- paste0("RpTools$rp_control(", arg.string, ")")
+  PEcAn.logger::logger.debug(paste0("Remote module executing the following function:\n", cmdFcn))
+  
+  # call rp_control
+  output <- do.call(RpTools$rp_control, fcn.args)
+  
+  # inserting data in the DB
+  if (!is.null(out_process_data)) {
+    # if the requested processed file already exists within the required timeline dont insert or update the DB
+    if (as.character(write_pro_start) == "dont write" &&
+        as.character(write_pro_end) == "dont write") {
```

Comment:
Is there a reason why `TRUE` and `FALSE` wouldn't work here instead of "dont write"? Or can't this case be assigned to another flag number?

Reply:
As documented in L18, these "write" variables are not being used as flags, they contain the dates which has to be written in the DB. This date might be different from what must be sent to `rp_control` hence I used separate variables for these. They would contain "don't write" when the requested data is already present and the DB dates don't have to be changed. I think I can use FALSE if that is preferable.